packages/wrangler/src/__tests__/pages.test.ts

```diff
@@ -1077,16 +1077,47 @@ describe("pages", () => {
 			setMockResponse(
 				"/accounts/:accountId/pages/projects/foo/deployments",
 				async ([_url, accountId], init) => {
-					assertLater(() => {
+					assertLater(async () => {
 						expect(accountId).toEqual("some-account-id");
 						expect(init.method).toEqual("POST");
 						const body = init.body as FormData;
 						const manifest = JSON.parse(body.get("manifest") as string);
+
+						// for Functions projects, we auto-generate a `_worker.js` and `_routes.json`
+						// file, based on the contents of `/functions`
+						const generatedWorkerJS = body.get("_worker.js") as Blob;
+						const generatedRoutesJSON = await (
+							body.get("_routes.json") as Blob
+						).text();
+
+						// make sure this is all we uploaded
+						expect([...body.keys()]).toEqual(["manifest", "_worker.js", "_routes.json"]);
+
 						expect(manifest).toMatchInlineSnapshot(`
 				                          Object {
 				                            "/README.md": "13a03eaf24ae98378acd36ea00f77f2f",
 				                          }
 			                      `);
+
+						// the contents of the generated `_worker.js` file is pretty massive, so I don't
```

Comment:
@GregBrimble pls let me know if you have other thoughts on this

Reply:
Yeah, what you have there is perfect. Thank you!